test/extended/prometheus/prometheus.go

```diff
@@ -49,6 +52,42 @@ type TelemeterClientConfig struct {
 	Enabled *bool `json:"enabled"`
 }
 
+var _ = g.Describe("[sig-instrumentation][Late] OpenShift service monitors [apigroup:image.openshift.io]", func() {
+	defer g.GinkgoRecover()
+	oc := exutil.NewCLIWithoutNamespace("prometheus")
+
+	g.It("should not be accessible without authorization", func() {
+		var errs []error
+		client, err := prometheusoperatorv1client.NewForConfig(oc.AdminConfig())
+		if err != nil {
+			g.Fail(fmt.Sprintf("failed to create monitoring client: %v", err))
+		}
+		serviceMonitorList, err := client.ServiceMonitors("").List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			g.Fail(fmt.Sprintf("failed to list service monitors: %v", err))
+		}
+		for _, sm := range serviceMonitorList.Items {
+			ok, err := endpointsNeedAuthorization(sm)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("failed to access service monitor %s/%s endpoints", sm.Namespace, sm.Name))
+			}
+			if !ok {
+				errs = append(errs, fmt.Errorf("service monitor %s/%s endpoints are accessible without authorization", sm.Namespace, sm.Name))
+			}
+		}
+		if err := kerrors.NewAggregate(errs); err != nil {
+			e2e.Fail(err.Error())
+		}
+	})
+
+})
+
+// endpointsNeedAuthorization return true is each endpoint of the service monitor responds 401 without authorization.
+// it raise an error if unexpected error occurs.
```

Comment:
```suggestion
// it raises an error if unexpected error occurs.
```

Reply:
The line no longer exists after refactoring.